File: OPENgovTIA/Generali/BasePage.vb
Imports System.Web.Security
Imports log4net
''' <summary>
''' Classe generale che eredita BasePage.
''' Viene inclusa in tutti i form perché contiene le funzioni basi che tutti i form devono avere.
''' </summary>
''' <remarks>In ottemperanza alle linee guida di sviluppo 1.0</remarks>
Public Class BaseEnte
    Inherits BasePage
    Private Shared Log As ILog = LogManager.GetLogger(GetType(BaseEnte))

    Private Sub BaseEnte_Init(sender As Object, e As EventArgs) Handles Me.Init
        Try
            If ConstSession.IdEnte = "" Then
                RegisterScript("GestAlert('a', 'warning', 'CmdLogOut', '', 'Sessione scaduta rieffettuare LOGIN');", Me.GetType)
            End If
        Catch ex As Exception
            Log.Debug(ConstSession.IdEnte +"."+ ConstSession.UserName + " - OPENgovTIA.BaseEnte.BasePage_Init.errore: ", ex)
        End Try
    End Sub
End Class
''' <summary>
''' Classe generale che eredita Page.
''' Viene inclusa in tutti i form perché contiene le funzioni basi che tutti i form devono avere.
''' </summary>
''' <remarks>In ottemperanza alle linee guida di sviluppo 1.0</remarks>
Public Class BasePage
    Inherits Page
    Private Shared Log As ILog = LogManager.GetLogger(GetType(BasePage))
    Public bp_TipoRuolo As String = ""
    Public Shared authCookie As HttpCookie

    Private Sub BasePage_Init(sender As Object, e As EventArgs) Handles Me.Init
        Session("COD_TRIBUTO") = Utility.Costanti.TRIBUTO_TARSU
        Session("IsFromTARES") = "1"
        Try
            Dim isExpired As Boolean = False
            'get cookie
            authCookie = HttpContext.Current.Request.Cookies.Get("aplckute")
            If Not authCookie Is Nothing Then
                Dim authTicket As FormsAuthenticationTicket = FormsAuthentication.Decrypt(authCookie.Value)
                If Not authTicket Is Nothing Then
                    If authTicket.Expired = False Then
                        Dim myData As String() = authTicket.UserData.Split("|")
                        If Not myData Is Nothing Then
                            If myData.GetUpperBound(0) <> 1 Then
                                Log.Debug("sessione scaduta perchè non ho userdata nel cookie")
                                isExpired = True
                            Else
                                'If myData(0) <> ConstSession.UserName Then
                                Log.Debug("utente cookie->" + myData(0) + "  utente sessione->" + ConstSession.UserName)
                                If ConstSession.UserName <> "" Then
                                    'Log.Debug("sessione scaduta perchè utente cookie diverso da utente sessione")
                                    'isExpired = True
                                Else
                                    Session("username") = myData(0)
                                    End If
                                'End If
                            End If
                        Else
                            Log.Debug("sessione scaduta perchè userdata cookie null")
                            isExpired = True
                        End If
                    Else
                        Log.Debug("sessione scaduta perchè cookie scaduto")
                        isExpired = True
                    End If
                Else
                    Log.Debug("sessione scaduta perchè non ticket cookie")
                    isExpired = True
                End If
            Else
                Log.Debug("sessione scaduta perchè non cookie")
                isExpired = True
            End If
            If ConstSession.UserName = "" Then
                Log.Debug("sessione scaduta perchè sessione operatore vuoto")
                isExpired = True
            End If
            If isExpired = True Then
                HttpContext.Current.Session("username") = ""
                RegisterScript("GestAlert('a', 'warning', 'CmdLogOut', '', 'Sessione scaduta rieffettuare LOGIN');", Me.GetType)
            End If
            If ConstSession.Ambiente.ToUpper() = "CMGC" Then
                If ConstSession.IsFromVariabile("1") Then
                End If
            End If
        Catch ex As Exception
            Log.Debug(ConstSession.IdEnte +"."+ ConstSession.UserName + " - OPENgovTIA.BasePage_Init.errore: ", ex)
        End Try
    End Sub

    Protected Sub RegisterScript(ByVal script As String, ByVal type As Type)
        ConstSession.CountScript = (ConstSession.CountScript + 1)
        Dim uniqueId As String = ("spc_" _
                    + (ConstSession.CountScript.ToString _
                    + (DateTime.Now.ToString + ("." + DateTime.Now.Millisecond.ToString))))
        Dim sScript As String = "<script language='javascript'>"
        sScript = (sScript + script)
        sScript = (sScript + "</script>")
        ClientScript.RegisterStartupScript(type, uniqueId, sScript)
    End Sub
End Class

File: OPENgovICI/Business/Basepage.cs
using System;
using System.Web.UI;
using log4net;
using System.Web.Security;
using System.Web;

namespace DichiarazioniICI
{
    /// <summary>
    /// Classe generale che eredita BasePage.
    /// Viene inclusa in tutti i form perché contiene le funzioni basi che tutti i form devono avere.
    /// </summary>
    /// <remarks>In ottemperanza alle linee guida di sviluppo 1.0</remarks>
    public class BaseEnte : BasePage
    {
        private static readonly ILog log = LogManager.GetLogger(typeof(BaseEnte));
        /// <summary>
        /// 
        /// </summary>
        /// <param name="e"></param>
        override protected void OnInit(EventArgs e)
        {
            try
            {
                if (Business.ConstWrapper.CodiceEnte == "" && Business.ConstWrapper.FromSovracomunali!="S" && Business.ConstWrapper.Sportello!="1")
                {
                    RegisterScript("GestAlert('a', 'warning', 'CmdLogOut', '', 'Sessione scaduta rieffettuare LOGIN');", this.GetType());
                }
            }
            catch (Exception Err)
            {
                log.Debug(Business.ConstWrapper.CodiceEnte + "."+ Business.ConstWrapper.sUsername + " - DichiarazioniICI.BaseEnte.OnInit.errore: ", Err);
            }
        }
    }
    /// <summary>
    /// Classe generale che eredita Page.
    /// Viene inclusa in tutti i form perché contiene le funzioni basi che tutti i form devono avere.
    /// </summary>
    /// <remarks>In ottemperanza alle linee guida di sviluppo 1.0</remarks>
    public class BasePage : Page
    {
        private static readonly ILog log = LogManager.GetLogger(typeof(BasePage));
        /// <summary>
        /// 
        /// </summary>
        public HttpCookie authCookie = null;
        /// <summary>
        /// 
        /// </summary>
        /// <param name="e"></param>
        override protected void OnInit(EventArgs e)
        {
            Session["COD_TRIBUTO"] = Utility.Costanti.TRIBUTO_ICI;
            try
            {
                bool isExpired = false;
                //get cookie
                authCookie = HttpContext.Current.Request.Cookies.Get("aplckute");
                if (authCookie != null)
                {
                    FormsAuthenticationTicket authTicket = FormsAuthentication.Decrypt(authCookie.Value);
                    if (authTicket != null)
                    {
                        if (!authTicket.Expired)
                        {
                            string[] myData = authTicket.UserData.Split(char.Parse("|"));
                            if (myData != null)
                            {
                                if (myData.GetUpperBound(0) != 1)
                                {
                                    log.Debug("sessione scaduta perchè non ho userdata nel cookie");
                                    isExpired = true;
                                }
                                else {
                                    //if (myData[0] != Business.ConstWrapper.sUsername)
                                    //{
                                    log.Debug("utente cookie->" + myData[0] + "  utente sessione->" + Business.ConstWrapper.sUsername);
                                  if (Business.ConstWrapper.sUsername != string.Empty)
                                        {
                                            //log.Debug("sessione scaduta perchè utente cookie diverso da utente sessione");
                                            //isExpired = true;
                                        }
                                        else {
                                            Session["username"] = myData[0];
                                        }
                                    //}
                                }
                            }
                            else {
                                log.Debug("sessione scaduta perchè userdata cookie null");
                                isExpired = true;
                            }
                        }
                        else {
                            log.Debug("sessione scaduta perchè cookie scaduto");
                            isExpired = true;
                        }
                    }
                    else {
                        log.Debug("sessione scaduta perchè non ticket cookie");
                        isExpired = true;
                    }
                }
                else {
                    if (Utility.StringOperation.FormatString(Business.ConstWrapper.Sportello) != "1")
                    {
                        log.Debug("sessione scaduta perchè non ho cookie");
                        isExpired = true;
                    }
                    else
                    {
                        Session["username"] = Business.ConstWrapper.SportelloUser;
                    }
                }
                if (Business.ConstWrapper.sUsername == "")
                {
                    log.Debug("sessione scaduta perchè sessione operatore vuoto");
                    isExpired = true;
                }
                if (isExpired)
                {
                    HttpContext.Current.Session["username"] = "";
                    RegisterScript("GestAlert('a', 'warning', 'CmdLogOut', '', 'Sessione scaduta rieffettuare LOGIN');", this.GetType());
                }
            }
            catch (Exception ex)
            {
                log.Debug(Business.ConstWrapper.CodiceEnte + "."+ Business.ConstWrapper.sUsername + " - DichiarazioniICI.BasePage_Init.errore: ", ex);
            }
        }
        /// <summary>
        /// 
        /// </summary>
        /// <param name="script"></param>
        /// <param name="type"></param>
        protected void RegisterScript(string script, Type type)
        {
            Business.ConstWrapper.CountScript = (Business.ConstWrapper.CountScript + 1);
            string uniqueId = "spc_" + Business.ConstWrapper.CountScript.ToString() + DateTime.Now.ToString() + "." + DateTime.Now.Millisecond.ToString();
            string sScript = "<script language='javascript'>";
            sScript += script;
            sScript += "</script>";
            ClientScript.RegisterStartupScript(type, uniqueId, sScript);
        }
    }
}